pyvista/core/datasetattributes.py

```diff
@@ -266,6 +650,25 @@ def remove(self, key: Union[int, str]) -> None:
         ----------
         key : int, str
             The name or index of the array to remove.
+
+        Examples
+        --------
+        Add a point data array to a DataSet and then remove it
+
+        >>> import pyvista
+        >>> mesh = pyvista.Cube().clean()
+        >>> mesh.point_arrays['my_data'] = range(mesh.n_points)
+        >>> mesh.point_arrays.remove('my_data')
+
+        Show that the array no longer exists in ``point_arrays``
+
+        >>> 'my_data' in mesh.point_arrays
+        False
+
+        Notes
+        -----
+        This is provided as VTK supports indexed arrays in DataSetAttributes.
```

Comment:
We should consider deprecating the usage of keys altogether and remove `list` like methods.  We should pick the class we're trying to represent and go with that.

What say you @adeak?  I know you had some issue with this as well?

Reply:
@akaszynski I still can't remember this coming up for me earlier. Do you perhaps mean the similar MultiBlock issue? https://github.com/pyvista/pyvista/issues/1507

Reply:
Perhaps that's what I was thinking of.  Regardless, deprecating this.

Reply:
Yeah, replacing almost-dicts with exactly-dicts (in terms of an API) always sounds good.

Reply:
I'm going to leave int in the `get_array` method since we still need this in some rare cases (as in the case of null array names).

However `__getitem__` and `__setitem__` both need to have their inputs match, so it makes sense to remove int indexing there.

Reply:
Sounds good. I'd have even went so far as to add a `get_array_by_index` method for those who need it (after all python dicts have been insertion-ordered for a while), but keeping the existing behaviour is even better.